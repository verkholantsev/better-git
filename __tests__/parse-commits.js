// @flow

import parseCommits from '../src/parse-commits';

describe('parseCommits()', () => {
    describe('should throw an error', () => {
        it('if there is unexpected line in input', () => {
            const input = 'bla';
            expect(() => parseCommits(input)).toThrow('Unexpected line: bla');
        });

        it('if there is unexpected line in `diff ...` line', () => {
            const input = 'diff bla';
            expect(() => parseCommits(input)).toThrow("Unexpected line 'diff bla' in parseChangedFile()");
        });

        it('if input does not contain initial `commit ...` line', () => {
            const input = 'Merge: abcd efgh';
            expect(() => parseCommits(input)).toThrow('Can not assign to last commit because commit list is empty');
        });
    });

    describe('should return correct output', () => {
        it('if input contains several commits', () => {
            const input = `commit a
diff --git a/one-file b/second-file
bla-bla

commit b
diff --git a/third-file b/fourth-file
bla-bla`;

            expect(parseCommits(input)).toMatchSnapshot();
        });
    });
});
