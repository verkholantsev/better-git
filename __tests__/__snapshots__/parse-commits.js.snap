// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseCommits() should return correct output if input contains several commits 1`] = `
Array [
  Object {
    "changedFiles": Array [
      Object {
        "destination": "second-file",
        "source": "one-file",
      },
    ],
    "commit": "a",
    "diff": "diff --git a/one-file b/second-file
bla-bla",
  },
  Object {
    "changedFiles": Array [
      Object {
        "destination": "fourth-file",
        "source": "third-file",
      },
    ],
    "commit": "b",
    "diff": "diff --git a/third-file b/fourth-file
bla-bla",
  },
]
`;
